crates/aiken-lang/src/gen_uplc.rs

```diff
@@ -4495,12 +4503,16 @@ impl<'a> CodeGenerator<'a> {
 
                 Some(term)
             }
-            Air::CastFromData { tipo } => {
+            Air::CastFromData { tipo, .. } => {
                 let mut term = arg_stack.pop().unwrap();
 
-                if extract_constant(&term).is_some() {
-                    term = builder::convert_data_to_type(term, &tipo);
+                term = if error_term == Term::Error {
+                    builder::convert_data_to_type(term, &tipo)
+                } else {
+                    builder::convert_data_to_type_debug(term, &tipo, error_term)
+                };
```

Comment:
If you define:

```rs
        let convert_data_to_type = |term, tipo| {
            if error_term == Term::Error {
                builder::convert_data_to_type(term, tipo)
            } else {
                builder::convert_data_to_type_debug(term, tipo, error_term.clone())
            }
        };
```

at the top of the of the `gen_uplc` function, you can then simplify this call and the two other to simply be:

```
convert_data_to_type(term, &tipo);
```

and avoid the repetition.